### PR TITLE
Add dynamic metadata support for ORIGINAL_DST cluster to pick a host

### PR DESCRIPTION
**Pull request by chabster4g**
Commit Message: Add dynamic metadata support for ORIGINAL_DST cluster to pick a host
Additional Description: One can specify a MetadataKey with a path selector to pick up a host from the dynamic metadata of the request. Selected value can either be a string or a list with at least a single element of string type.
Risk Level: Low
Testing: Unit Testing
Docs Changes: N/A
Release Notes: 
Platform Specific Features:

### CODE REVIEW

**Review by yanavlasov — APPROVED** [review]

**Comment by yanavlasov** [comment]
LGTM, waiting for the API approval.

**Review by mattklein123 — CHANGES_REQUESTED** [review]
Please add a release note, thank you. Otherwise LGTM.

/wait

File: changelogs/current.yaml
Comment (mattklein123) [inline changelogs/current.yaml:437]: Please ref link to the new field.

/wait

**Review by yanavlasov — APPROVED** [review]

File: changelogs/current.yaml
Comment (chabster4g) [reply changelogs/current.yaml]: Done

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**
